Add configurable Prometheus PodMonitor resource

dpunkturban opened the pull request:
This PR adds the Prometheus PodMonitor as configurable resource, so you don't need to deploy the PodMonitor extra.

Please ensure your pull request adheres to the following guidelines:
- [x] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [x] Document your Changes in the `CHANGELOG.md` file of the respected chart as well as the `Chart.yaml`
- [x] Bump the Version number of the respected chart

mydoomfr commented:
Hello,

I believe it's an excellent idea to provide complete integration with Prometheus. Also, would it be a good idea to add /api/monitoring/metrics endpoint using the monitoring passcode to authenticate?

```
  - interval: 30s
    path: /api/monitoring/metrics
    scheme: http
    port: http
    bearerTokenSecret:
      name: {{ template "sonarqube.fullname" . }}-monitoring-passcode
      key: SONAR_WEB_SYSTEMPASSCODE
```

achupryn commented on charts/sonarqube/values.yaml:
```diff
@@ -258,6 +258,21 @@ prometheusExporter:
+  #
+  podMonitor:
+    # Create PodMonitor Resource for Prometheus scraping
+    enabled: true
```
podMonitor should be disabled by default, because it requires prometheus-operator installed.
The same for sonarqube-dce/lts

dpunkturban replied:
Hi,

the podMonitor template is checks if prometheus and podmonitor is enabeld, so we never deploy the monitor without prometheus. 

```
{{- if and .Values.prometheusExporter.enabled .Values.prometheusExporter.podMonitor.enabled }}
```

I can change the default if you want :)

dpunkturban replied:
Ah got your point, yes i change the default! sorry

dpunkturban replied:
fixed :)

jCOTINEAU commented:
Hello @dpunkturban thanks a lot for taking the time to improve our SonarQube helm chart.

We will try to review and test it in detail during the following weeks.

The initial comment i can make is that the LTS chart is frozen as per our LTS policy, we modify it mostly only for security reason. Could you please remove everything related to the LTS ?

Thanks a lot again for participating in the community !

jCOTINEAU commented:
Hello @dpunkturban, any update on this PR ?

In order for it to be tested and merge, you should rebase from master, remove the LTS related stuff and address @achupryn point.

Then we will be able to validate everything and merge it.

Thanks a lot for taking time to participate in the community !

dpunkturban commented:
Hi @jCOTINEAU , sorry for the delay, was quite busy in the latest Q4 Weeks. 

I would finish the PR this weekend :)

dpunkturban commented:
HI @jCOTINEAU,

- Rebased with Master 
- Removed LTS changed
- Added `/api/monitoring/credentials` target
- Set PodMonitor default to false (because you need Prometheus operator for this)

Half Off-Topic:
In the monitoring passcode secret there is this if-statement everywhere, where values are harcoded, if they are not defined in the values:

```
{{- if and .Values.monitoringPasscodeSecretName .Values.monitoringPasscodeSecretKey }}
      name: {{ .Values.monitoringPasscodeSecretName }}
      key: {{ .Values.monitoringPasscodeSecretKey }}
    {{- else }}
      name: {{ template "sonarqube.fullname" . }}-monitoring-passcode
      key: SONAR_WEB_SYSTEMPASSCODE
```

Why are these values not set directly in the values.yaml? Then we could delete the if-statement and would save some copy&paste code.

```
monitoringPasscodeSecretName: monitoring-passcode
monitoringPasscodeSecretKey: SONAR_WEB_SYSTEMPASSCODE
```

I did not want to modify more for the PR now, so just the question and not directly a commit for it :)

Thanks
Dennis

jCOTINEAU commented:
Hello @dpunkturban i just want to give you an update on that PR.

We have a new chart LTS release coming up.
Which is the reason why we are not yet incorporating your changes.

I deeply thank you for taking the time to participate in the community regarding the time it takes for the PR to be accepted.

I would nonetheless like to highlight from our [CONTRIBUTION.md](https://github.com/SonarSource/helm-chart-sonarqube#contributing) that it is hard for us to deal with external PR and we always prefer feature demand on our community forum.

Huge thanks again, may you excuse us for the inconvenience

jCOTINEAU commented:
Hello @dpunkturban 

Thanks again for your contribution. As said before, we will consider your change after the LTS is released, in one of the next hardening sprints.

I have just written a small task about it ([SQHELM-107](https://sonarsource.atlassian.net/browse/SQHELM-107)) and we'll make sure to tackle it using the work you did on this feature branch.

For now, I'm closing this PR.
